### ISSUE TITLE
Cilium agent pod got crashloopbackoff after upgrade to bullseye

### ISSUE THREAD

**Issue by dispensable** [cilium/cilium#21470]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Cilium agent pod keeps crashing on bullseye node.

### Cilium Version

Client: 1.12.2 c7516b9 2022-09-14T15:25:06+02:00 go version go1.18.6 linux/amd64
Daemon: 1.12.2 c7516b9 2022-09-14T15:25:06+02:00 go version go1.18.6 linux/amd64

### Kernel Version

Linux bullseye1 5.10.0-17-amd64 #1 SMP Debian 5.10.136-1 (2022-08-13) x86_64 GNU/Linux

### Kubernetes Version

v1.22.4

### Sysdump

The pod keeps crashing and I just can't run any cmd in it.

### Relevant log output

```shell
level=debug msg="Skipped reading configuration file" error="Config File \"ciliumd\" Not Found in \"[/root]\"" subsys=config
level=info msg="Started gops server" address="127.0.0.1:9890" subsys=daemon
level=info msg="Memory available for map entries (0.003% of 8075042816B): 20187607B" subsys=config
level=debug msg="Total memory for default map entries: 149422080" subsys=config
level=info msg="option bpf-ct-global-tcp-max set by dynamic sizing to 131072" subsys=config
level=info msg="option bpf-ct-global-any-max set by dynamic sizing to 65536" subsys=config
level=info msg="option bpf-nat-global-max set by dynamic sizing to 131072" subsys=config
level=info msg="option bpf-neigh-global-max set by dynamic sizing to 131072" subsys=config
level=info msg="option bpf-sock-rev-map-max set by dynamic sizing to 65536" subsys=config
level=info msg="  --agent-health-port='9879'" subsys=daemon
level=info msg="  --agent-labels=''" subsys=daemon
level=info msg="  --agent-not-ready-taint-key='node.cilium.io/agent-not-ready'" subsys=daemon
level=info msg="  --allocator-list-timeout='3m0s'" subsys=daemon
level=info msg="  --allow-icmp-frag-needed='true'" subsys=daemon
level=info msg="  --allow-localhost='auto'" subsys=daemon
level=info msg="  --annotate-k8s-node='false'" subsys=daemon
level=info msg="  --api-rate-limit=''" subsys=daemon
level=info msg="  --arping-refresh-period='30s'" subsys=daemon
level=info msg="  --auto-create-cilium-node-resource='true'" subsys=daemon
level=info msg="  --auto-direct-node-routes='false'" subsys=daemon
level=info msg="  --bgp-announce-lb-ip='false'" subsys=daemon
level=info msg="  --bgp-announce-pod-cidr='false'" subsys=daemon
level=info msg="  --bgp-config-path='/var/lib/cilium/bgp/config.yaml'" subsys=daemon
level=info msg="  --bpf-ct-global-any-max='262144'" subsys=daemon
level=info msg="  --bpf-ct-global-tcp-max='524288'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-fin='10s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-syn='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp-grace='1m0s'" subsys=daemon
level=info msg="  --bpf-filter-priority='1'" subsys=daemon
level=info msg="  --bpf-fragments-map-max='8192'" subsys=daemon
level=info msg="  --bpf-lb-acceleration='disabled'" subsys=daemon
level=info msg="  --bpf-lb-affinity-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-algorithm='random'" subsys=daemon
level=info msg="  --bpf-lb-dev-ip-addr-inherit=''" subsys=daemon
level=info msg="  --bpf-lb-dsr-dispatch='opt'" subsys=daemon
level=info msg="  --bpf-lb-dsr-l4-xlate='frontend'" subsys=daemon
level=info msg="  --bpf-lb-external-clusterip='false'" subsys=daemon
level=info msg="  --bpf-lb-maglev-hash-seed='JLfvgnHc2kaSUFaI'" subsys=daemon
level=info msg="  --bpf-lb-maglev-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-maglev-table-size='16381'" subsys=daemon
level=info msg="  --bpf-lb-map-max='65536'" subsys=daemon
level=info msg="  --bpf-lb-mode='snat'" subsys=daemon
level=info msg="  --bpf-lb-rev-nat-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv4-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-rss-ipv6-src-cidr=''" subsys=daemon
level=info msg="  --bpf-lb-service-backend-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-service-map-max='0'" subsys=daemon
level=info msg="  --bpf-lb-sock='false'" subsys=daemon
level=info msg="  --bpf-lb-sock-hostns-only='false'" subsys=daemon
level=info msg="  --bpf-lb-source-range-map-max='0'" subsys=daemon
level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon
level=info msg="  --bpf-nat-global-max='524288'" subsys=daemon
level=info msg="  --bpf-neigh-global-max='524288'" subsys=daemon
level=info msg="  --bpf-policy-map-max='16384'" subsys=daemon
level=info msg="  --bpf-root='/sys/fs/bpf'" subsys=daemon
level=info msg="  --bpf-sock-rev-map-max='262144'" subsys=daemon
level=info msg="  --bypass-ip-availability-upon-restore='false'" subsys=daemon
level=info msg="  --certificates-directory='/var/run/cilium/certs'" subsys=daemon
level=info msg="  --cflags=''" subsys=daemon
level=info msg="  --cgroup-root='/run/cilium/cgroupv2'" subsys=daemon
level=info msg="  --cluster-health-port='4240'" subsys=daemon
level=info msg="  --cluster-id='0'" subsys=daemon
level=info msg="  --cluster-name='default'" subsys=daemon
level=info msg="  --clustermesh-config='/var/lib/cilium/clustermesh/'" subsys=daemon
level=info msg="  --cmdref=''" subsys=daemon
level=info msg="  --cni-chaining-mode=''" subsys=daemon
level=info msg="  --config=''" subsys=daemon
level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=daemon
level=info msg="  --conntrack-gc-interval='0s'" subsys=daemon
level=info msg="  --crd-wait-timeout='5m0s'" subsys=daemon
level=info msg="  --datapath-mode='veth'" subsys=daemon
level=info msg="  --debug='true'" subsys=daemon
level=info msg="  --debug-verbose=''" subsys=daemon
level=info msg="  --derive-masquerade-ip-addr-from-device=''" subsys=daemon
level=info msg="  --devices=''" subsys=daemon
level=info msg="  --direct-routing-device=''" subsys=daemon
level=info msg="  --disable-cnp-status-updates='true'" subsys=daemon
level=info msg="  --disable-conntrack='false'" subsys=daemon
level=info msg="  --disable-endpoint-crd='false'" subsys=daemon
level=info msg="  --disable-envoy-version-check='false'" subsys=daemon
level=info msg="  --disable-iptables-feeder-rules=''" subsys=daemon
level=info msg="  --dns-max-ips-per-restored-rule='1000'" subsys=daemon
level=info msg="  --dns-policy-unload-on-shutdown='false'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-limit='0'" subsys=daemon
level=info msg="  --dnsproxy-concurrency-processing-grace-period='0s'" subsys=daemon
level=info msg="  --egress-masquerade-interfaces=''" subsys=daemon
level=info msg="  --egress-multi-home-ip-rule-compat='false'" subsys=daemon
level=info msg="  --enable-auto-protect-node-port-range='true'" subsys=daemon
level=info msg="  --enable-bandwidth-manager='false'" subsys=daemon
level=info msg="  --enable-bbr='false'" subsys=daemon
level=info msg="  --enable-bgp-control-plane='false'" subsys=daemon
level=info msg="  --enable-bpf-clock-probe='true'" subsys=daemon
level=info msg="  --enable-bpf-masquerade='true'" subsys=daemon
level=info msg="  --enable-bpf-tproxy='false'" subsys=daemon
level=info msg="  --enable-cilium-endpoint-slice='false'" subsys=daemon
level=info msg="  --enable-custom-calls='false'" subsys=daemon
level=info msg="  --enable-endpoint-health-checking='true'" subsys=daemon
level=info msg="  --enable-endpoint-routes='false'" subsys=daemon
level=info msg="  --enable-envoy-config='false'" subsys=daemon
level=info msg="  --enable-external-ips='true'" subsys=daemon
level=info msg="  --enable-health-check-nodeport='true'" subsys=daemon
level=info msg="  --enable-health-checking='true'" subsys=daemon
level=info msg="  --enable-host-firewall='false'" subsys=daemon
level=info msg="  --enable-host-legacy-routing='false'" subsys=daemon
level=info msg="  --enable-host-port='true'" subsys=daemon
level=info msg="  --enable-host-reachable-services='false'" subsys=daemon
level=info msg="  --enable-hubble='true'" subsys=daemon
level=info msg="  --enable-hubble-recorder-api='true'" subsys=daemon
level=info msg="  --enable-icmp-rules='true'" subsys=daemon
level=info msg="  --enable-identity-mark='true'" subsys=daemon
level=info msg="  --enable-ip-masq-agent='false'" subsys=daemon
level=info msg="  --enable-ipsec='false'" subsys=daemon
level=info msg="  --enable-ipv4='true'" subsys=daemon
level=info msg="  --enable-ipv4-egress-gateway='false'" subsys=daemon
level=info msg="  --enable-ipv4-fragment-tracking='true'" subsys=daemon
level=info msg="  --enable-ipv4-masquerade='true'" subsys=daemon
level=info msg="  --enable-ipv6='false'" subsys=daemon
level=info msg="  --enable-ipv6-masquerade='true'" subsys=daemon
level=info msg="  --enable-ipv6-ndp='false'" subsys=daemon
level=info msg="  --enable-k8s-api-discovery='false'" subsys=daemon
level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=daemon
level=info msg="  --enable-k8s-event-handover='false'" subsys=daemon
level=info msg="  --enable-k8s-terminating-endpoint='true'" subsys=daemon
level=info msg="  --enable-l2-neigh-discovery='true'" subsys=daemon
level=info msg="  --enable-l7-proxy='true'" subsys=daemon
level=info msg="  --enable-local-node-route='true'" subsys=daemon
level=info msg="  --enable-local-redirect-policy='false'" subsys=daemon
level=info msg="  --enable-mke='false'" subsys=daemon
level=info msg="  --enable-monitor='true'" subsys=daemon
level=info msg="  --enable-node-port='false'" subsys=daemon
level=info msg="  --enable-policy='default'" subsys=daemon
level=info msg="  --enable-recorder='false'" subsys=daemon
level=info msg="  --enable-remote-node-identity='true'" subsys=daemon
level=info msg="  --enable-runtime-device-detection='false'" subsys=daemon
level=info msg="  --enable-selective-regeneration='true'" subsys=daemon
level=info msg="  --enable-service-topology='false'" subsys=daemon
level=info msg="  --enable-session-affinity='false'" subsys=daemon
level=info msg="  --enable-svc-source-range-check='true'" subsys=daemon
level=info msg="  --enable-tracing='false'" subsys=daemon
level=info msg="  --enable-unreachable-routes='false'" subsys=daemon
level=info msg="  --enable-vtep='false'" subsys=daemon
level=info msg="  --enable-well-known-identities='false'" subsys=daemon
level=info msg="  --enable-wireguard='false'" subsys=daemon
level=info msg="  --enable-wireguard-userspace-fallback='false'" subsys=daemon
level=info msg="  --enable-xdp-prefilter='false'" subsys=daemon
level=info msg="  --enable-xt-socket-fallback='true'" subsys=daemon
level=info msg="  --encrypt-interface=''" subsys=daemon
level=info msg="  --encrypt-node='false'" subsys=daemon
level=info msg="  --endpoint-gc-interval='5m0s'" subsys=daemon
level=info msg="  --endpoint-interface-name-prefix=''" subsys=daemon
level=info msg="  --endpoint-queue-size='25'" subsys=daemon
level=info msg="  --endpoint-status=''" subsys=daemon
level=info msg="  --envoy-config-timeout='2m0s'" subsys=daemon
level=info msg="  --envoy-log=''" subsys=daemon
level=info msg="  --exclude-local-address=''" subsys=daemon
level=info msg="  --fixed-identity-mapping=''" subsys=daemon
level=info msg="  --force-local-policy-eval-at-source='true'" subsys=daemon
level=info msg="  --fqdn-regex-compile-lru-size='1024'" subsys=daemon
level=info msg="  --gops-port='9890'" subsys=daemon
level=info msg="  --host-reachable-services-protos='tcp,udp'" subsys=daemon
level=info msg="  --http-403-msg=''" subsys=daemon
level=info msg="  --http-idle-timeout='0'" subsys=daemon
level=info msg="  --http-max-grpc-timeout='0'" subsys=daemon
level=info msg="  --http-normalize-path='true'" subsys=daemon
level=info msg="  --http-request-timeout='3600'" subsys=daemon
level=info msg="  --http-retry-count='3'" subsys=daemon
level=info msg="  --http-retry-timeout='0'" subsys=daemon
level=info msg="  --hubble-disable-tls='false'" subsys=daemon
level=info msg="  --hubble-event-buffer-capacity='4095'" subsys=daemon
level=info msg="  --hubble-event-queue-size='0'" subsys=daemon
level=info msg="  --hubble-export-file-compress='false'" subsys=daemon
level=info msg="  --hubble-export-file-max-backups='5'" subsys=daemon
level=info msg="  --hubble-export-file-max-size-mb='10'" subsys=daemon
level=info msg="  --hubble-export-file-path=''" subsys=daemon
level=info msg="  --hubble-listen-address=':4244'" subsys=daemon
level=info msg="  --hubble-metrics=''" subsys=daemon
level=info msg="  --hubble-metrics-server=''" subsys=daemon
level=info msg="  --hubble-recorder-sink-queue-size='1024'" subsys=daemon
level=info msg="  --hubble-recorder-storage-path='/var/run/cilium/pcaps'" subsys=daemon
level=info msg="  --hubble-socket-path='/var/run/cilium/hubble.sock'" subsys=daemon
level=info msg="  --hubble-tls-cert-file='/var/lib/cilium/tls/hubble/server.crt'" subsys=daemon
level=info msg="  --hubble-tls-client-ca-files='/var/lib/cilium/tls/hubble/client-ca.crt'" subsys=daemon
level=info msg="  --hubble-tls-key-file='/var/lib/cilium/tls/hubble/server.key'" subsys=daemon
level=info msg="  --identity-allocation-mode='crd'" subsys=daemon
level=info msg="  --identity-change-grace-period='5s'" subsys=daemon
level=info msg="  --identity-restore-grace-period='10m0s'" subsys=daemon
level=info msg="  --install-egress-gateway-routes='false'" subsys=daemon
level=info msg="  --install-iptables-rules='true'" subsys=daemon
level=info msg="  --install-no-conntrack-iptables-rules='false'" subsys=daemon
level=info msg="  --ip-allocation-timeout='2m0s'" subsys=daemon
level=info msg="  --ip-masq-agent-config-path='/etc/config/ip-masq-agent'" subsys=daemon
level=info msg="  --ipam='cluster-pool'" subsys=daemon
level=info msg="  --ipsec-key-file=''" subsys=daemon
level=info msg="  --iptables-lock-timeout='5s'" subsys=daemon
level=info msg="  --iptables-random-fully='false'" subsys=daemon
level=info msg="  --ipv4-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv4-node='auto'" subsys=daemon
level=info msg="  --ipv4-pod-subnets=''" subsys=daemon
level=info msg="  --ipv4-range='auto'" subsys=daemon
level=info msg="  --ipv4-service-loopback-address='169.254.42.1'" subsys=daemon
level=info msg="  --ipv4-service-range='auto'" subsys=daemon
level=info msg="  --ipv6-cluster-alloc-cidr='f00d::/64'" subsys=daemon
level=info msg="  --ipv6-mcast-device=''" subsys=daemon
level=info msg="  --ipv6-native-routing-cidr=''" subsys=daemon
level=info msg="  --ipv6-node='auto'" subsys=daemon
level=info msg="  --ipv6-pod-subnets=''" subsys=daemon
level=info msg="  --ipv6-range='auto'" subsys=daemon
level=info msg="  --ipv6-service-range='auto'" subsys=daemon
level=info msg="  --join-cluster='false'" subsys=daemon
level=info msg="  --k8s-api-server=''" subsys=daemon
level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=daemon
level=info msg="  --k8s-kubeconfig-path=''" subsys=daemon
level=info msg="  --k8s-namespace='kube-system'" subsys=daemon
level=info msg="  --k8s-require-ipv4-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-require-ipv6-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-service-cache-size='128'" subsys=daemon
level=info msg="  --k8s-service-proxy-name=''" subsys=daemon
level=info msg="  --k8s-sync-timeout='3m0s'" subsys=daemon
level=info msg="  --k8s-watcher-endpoint-selector='metadata.name!=kube-scheduler,metadata.name!=kube-controller-manager,metadata.name!=etcd-operator,metadata.name!=gcp-controller-manager'" subsys=daemon
level=info msg="  --keep-config='false'" subsys=daemon
level=info msg="  --kube-proxy-replacement='strict'" subsys=daemon
level=info msg="  --kube-proxy-replacement-healthz-bind-address=''" subsys=daemon
level=info msg="  --kvstore='etcd'" subsys=daemon
level=info msg="  --kvstore-connectivity-timeout='2m0s'" subsys=daemon
level=info msg="  --kvstore-lease-ttl='15m0s'" subsys=daemon
level=info msg="  --kvstore-max-consecutive-quorum-errors='2'" subsys=daemon
level=info msg="  --kvstore-opt='{\"etcd.config\":,\"/var/lib/etcd-config/etcd.config\"}'" subsys=daemon
level=info msg="  --kvstore-periodic-sync='5m0s'" subsys=daemon
level=info msg="  --label-prefix-file=''" subsys=daemon
level=info msg="  --labels=''" subsys=daemon
level=info msg="  --lib-dir='/var/lib/cilium'" subsys=daemon
level=info msg="  --local-max-addr-scope='252'" subsys=daemon
level=info msg="  --local-router-ipv4=''" subsys=daemon
level=info msg="  --local-router-ipv6=''" subsys=daemon
level=info msg="  --log-driver=''" subsys=daemon
level=info msg="  --log-opt=''" subsys=daemon
level=info msg="  --log-system-load='false'" subsys=daemon
level=info msg="  --max-controller-interval='0'" subsys=daemon
level=info msg="  --metrics=''" subsys=daemon
level=info msg="  --mke-cgroup-mount=''" subsys=daemon
level=info msg="  --monitor-aggregation='medium'" subsys=daemon
level=info msg="  --monitor-aggregation-flags='all'" subsys=daemon
level=info msg="  --monitor-aggregation-interval='5s'" subsys=daemon
level=info msg="  --monitor-queue-size='0'" subsys=daemon
level=info msg="  --mtu='0'" subsys=daemon
level=info msg="  --node-port-acceleration='disabled'" subsys=daemon
level=info msg="  --node-port-algorithm='random'" subsys=daemon
level=info msg="  --node-port-bind-protection='true'" subsys=daemon
level=info msg="  --node-port-mode='snat'" subsys=daemon
level=info msg="  --node-port-range='30000,32767'" subsys=daemon
level=info msg="  --policy-audit-mode='false'" subsys=daemon
level=info msg="  --policy-queue-size='100'" subsys=daemon
level=info msg="  --policy-trigger-interval='1s'" subsys=daemon
level=info msg="  --pprof='false'" subsys=daemon
level=info msg="  --pprof-port='6060'" subsys=daemon
level=info msg="  --preallocate-bpf-maps='false'" subsys=daemon
level=info msg="  --prepend-iptables-chains='true'" subsys=daemon
level=info msg="  --procfs='/host/proc'" subsys=daemon
level=info msg="  --prometheus-serve-addr=':9962'" subsys=daemon
level=info msg="  --proxy-connect-timeout='1'" subsys=daemon
level=info msg="  --proxy-gid='1337'" subsys=daemon
level=info msg="  --proxy-max-connection-duration-seconds='0'" subsys=daemon
level=info msg="  --proxy-max-requests-per-connection='0'" subsys=daemon
level=info msg="  --proxy-prometheus-port='0'" subsys=daemon
level=info msg="  --read-cni-conf=''" subsys=daemon
level=info msg="  --restore='true'" subsys=daemon
level=info msg="  --route-metric='0'" subsys=daemon
level=info msg="  --sidecar-istio-proxy-image='cilium/istio_proxy'" subsys=daemon
level=info msg="  --single-cluster-route='false'" subsys=daemon
level=info msg="  --socket-path='/var/run/cilium/cilium.sock'" subsys=daemon
level=info msg="  --sockops-enable='false'" subsys=daemon
level=info msg="  --state-dir='/var/run/cilium'" subsys=daemon
level=info msg="  --tofqdns-dns-reject-response-code='refused'" subsys=daemon
level=info msg="  --tofqdns-enable-dns-compression='true'" subsys=daemon
level=info msg="  --tofqdns-endpoint-max-ip-per-hostname='50'" subsys=daemon
level=info msg="  --tofqdns-idle-connection-grace-period='0s'" subsys=daemon
level=info msg="  --tofqdns-max-deferred-connection-deletes='10000'" subsys=daemon
level=info msg="  --tofqdns-min-ttl='3600'" subsys=daemon
level=info msg="  --tofqdns-pre-cache=''" subsys=daemon
level=info msg="  --tofqdns-proxy-port='0'" subsys=daemon
level=info msg="  --tofqdns-proxy-response-max-delay='100ms'" subsys=daemon
level=info msg="  --trace-payloadlen='128'" subsys=daemon
level=info msg="  --tunnel='vxlan'" subsys=daemon
level=info msg="  --tunnel-port='0'" subsys=daemon
level=info msg="  --version='false'" subsys=daemon
level=info msg="  --vlan-bpf-bypass=''" subsys=daemon
level=info msg="  --vtep-cidr=''" subsys=daemon
level=info msg="  --vtep-endpoint=''" subsys=daemon
level=info msg="  --vtep-mac=''" subsys=daemon
level=info msg="  --vtep-mask=''" subsys=daemon
level=info msg="  --write-cni-conf-when-ready=''" subsys=daemon
level=info msg="     _ _ _" subsys=daemon
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
level=info msg="|  _| | | | | |     |" subsys=daemon
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
level=info msg="Cilium 1.12.2 c7516b9 2022-09-14T15:25:06+02:00 go version go1.18.6 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: 48cbb2f88edca93ba80d1135dbb067a827b8adc2/1.21.5/Distribution/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (5.10.0) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Mounted cgroupv2 filesystem at /run/cilium/cgroupv2" subsys=cgroups
level=info msg="Parsing base label prefixes from default label list" subsys=labels-filter
level=info msg="Parsing additional label prefixes from user inputs: []" subsys=labels-filter
level=info msg="Final label prefixes to be used for identity evaluation:" subsys=labels-filter
level=info msg=" - reserved:.*" subsys=labels-filter
level=info msg=" - :io\\.kubernetes\\.pod\\.namespace" subsys=labels-filter
level=info msg=" - :io\\.cilium\\.k8s\\.namespace\\.labels" subsys=labels-filter
level=info msg=" - :app\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:io\\.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:.*beta\\.kubernetes\\.io" subsys=labels-filter
level=info msg=" - !:k8s\\.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Auto-disabling \"enable-bpf-clock-probe\" feature since KERNEL_HZ cannot be determined" error="Cannot probe CONFIG_HZ" subsys=daemon
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.97.0.0/16
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Establishing connection to apiserver" host="https://sa-borin.k8s.douban:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=debug msg="Starting new controller" name=k8s-heartbeat subsys=controller uuid=5b2003a3-a377-46f2-a46b-4540e00833aa
level=debug msg="Controller func execution time: 832ns" name=k8s-heartbeat subsys=controller uuid=5b2003a3-a377-46f2-a46b-4540e00833aa
level=debug msg="Skipping Leases support fallback discovery" subsys=k8s
level=debug msg="Starting new controller" name=endpoint-gc subsys=controller uuid=78033399-a4b5-4d5f-8a70-12340d59426b
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"bpf-lb-sock\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=10.1.32.97 mtu=1500 subsys=mtu
level=debug msg="creating new EventQueue" name=config-modify-queue numBufferedEvents=10 subsys=eventqueue
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ipcache subsys=bpf
level=debug msg="Controller func execution time: 1.022µs" name=endpoint-gc subsys=controller uuid=78033399-a4b5-4d5f-8a70-12340d59426b
level=debug msg="Performing regular background work" subsys=nodemanager syncInterval=1m0s
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ipcache subsys=bpf
level=debug msg="getting identity cache for identity allocator manager" subsys=identity-cache
level=debug msg="creating new EventQueue" name=repository-change-queue numBufferedEvents=100 subsys=eventqueue
level=debug msg="creating new EventQueue" name=repository-reaction-queue numBufferedEvents=100 subsys=eventqueue
level=info msg="Restoring 1 old CIDR identities" subsys=daemon
level=debug msg="Resolving identity" identityLabels="cidr:10.0.132.76/32,reserved:world" subsys=identity-cache
level=debug msg="Reallocated restored CIDR identity: 16777217" subsys=identity-cache
level=debug msg="UpdateIdentities: Adding a new identity" identity=16777217 labels="[cidr:0.0.0.0/0 cidr:0.0.0.0/1 cidr:0.0.0.0/2 cidr:0.0.0.0/3 cidr:0.0.0.0/4 cidr:10.0.0.0/10 cidr:10.0.0.0/11 cidr:10.0.0.0/12 cidr:10.0.0.0/13 cidr:10.0.0.0/14 cidr:10.0.0.0/15 cidr:10.0.0.0/16 cidr:10.0.0.0/7 cidr:10.0.0.0/8 cidr:10.0.0.0/9 cidr:10.0.128.0/17 cidr:10.0.128.0/18 cidr:10.0.128.0/19 cidr:10.0.128.0/20 cidr:10.0.128.0/21 cidr:10.0.132.0/22 cidr:10.0.132.0/23 cidr:10.0.132.0/24 cidr:10.0.132.0/25 cidr:10.0.132.64/26 cidr:10.0.132.64/27 cidr:10.0.132.64/28 cidr:10.0.132.72/29 cidr:10.0.132.76/30 cidr:10.0.132.76/31 cidr:10.0.132.76/32 cidr:8.0.0.0/5 cidr:8.0.0.0/6 reserved:world]" subsys=policy
level=debug msg="Regenerating all endpoints" subsys=policy
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lxc subsys=bpf
level=debug msg="Opening map in parallel mode" file-path=/sys/fs/bpf/tc/globals/cilium_ipcache name=cilium_ipcache subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ipcache subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_metrics subsys=ebpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_tunnel_map subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_reverse_sk subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_reverse_sk subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_services_v2 subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_backends_v2 subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_reverse_nat subsys=bpf
..........
........
level=debug msg="Adding backends to affinity match map" backends="[]" serviceID=342 subsys=service
level=debug msg="Kubernetes service definition changed" action=service-updated endpoints="10.1.32.100:10250/TCP,10.1.32.100:10255/TCP,10.1.32.100:4194/TCP,10.1.32.125:10250/TCP,10.1.32.125:10255/TCP,10.1.32.125:4194/TCP,10.1.32.128:10250/TCP,10.1.32.128:10255/TCP,10.1.32.128:4194/TCP,10.1.32.30:10250/TCP,10.1.32.30:10255/TCP,10.1.32.30:4194/TCP,10.1.32.40:10250/TCP,10.1.32.40:10255/TCP,10.1.32.40:4194/TCP,10.1.32.43:10250/TCP,10.1.32.43:10255/TCP,10.1.32.43:4194/TCP,10.1.32.65:10250/TCP,10.1.32.65:10255/TCP,10.1.32.65:4194/TCP,10.1.32.72:10250/TCP,10.1.32.72:10255/TCP,10.1.32.72:4194/TCP,10.1.32.77:10250/TCP,10.1.32.77:10255/TCP,10.1.32.77:4194/TCP,10.1.32.81:10250/TCP,10.1.32.81:10255/TCP,10.1.32.81:4194/TCP,10.1.32.89:10250/TCP,10.1.32.89:10255/TCP,10.1.32.89:4194/TCP,10.1.32.97:10250/TCP,10.1.32.97:10255/TCP,10.1.32.97:4194/TCP" k8sNamespace=kube-system k8sSvcName=kube-prom-kube-prometheus-kubelet old-service=nil service="frontends:[]/ports=[cadvisor https-metrics http-metrics]/selector=map[]" subsys=k8s-watcher
level=debug msg="Kubernetes service definition changed" action=service-updated endpoints="10.2.4.20:8012/TCP" k8sNamespace=sa-common k8sSvcName=lark-recorder-00001 old-service=nil service="frontends:[172.23.202.8]/ports=[http]/selector=map[]" subsys=k8s-watcher
level=debug msg="Upserting service" backends="[{http 0 borin6 {10.2.4.20 {TCP 8012} 0} 0 false false}]" l7LBFrontendPorts="[]" l7LBProxyPort=0 loadBalancerSourceRanges="[]" serviceIP="{172.23.202.8 {TCP 80} 0}" serviceName=lark-recorder-00001 serviceNamespace=sa-common sessionAffinity=false sessionAffinityTimeout=0 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=ClusterIP
level=debug msg="Acquired service ID" backends="[{http 0 borin6 {10.2.4.20 {TCP 8012} 0} 0 false false}]" l7LBFrontendPorts="[]" l7LBProxyPort=0 loadBalancerSourceRanges="[]" serviceID=254 serviceIP="{172.23.202.8 {TCP 80} 0}" serviceName=lark-recorder-00001 serviceNamespace=sa-common sessionAffinity=false sessionAffinityTimeout=0 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=ClusterIP
level=debug msg="Deleting backends from session affinity match" backends="[]" serviceID=254 subsys=service
level=debug msg="Upserted service entry" backendSlot=1 subsys=map-lb svcKey="172.23.202.8:20480" svcVal="1 0 (65024) [0x0 0x0]"
level=debug msg="Upserted service entry" backendSlot=0 subsys=map-lb svcKey="172.23.202.8:20480" svcVal="0 1 (65024) [0x0 0x0]"
level=debug msg="Adding backends to affinity match map" backends="[]" serviceID=254 subsys=service
level=debug msg="Kubernetes service definition changed" action=service-updated endpoints="10.2.4.20:8012/TCP" k8sNamespace=sa-common k8sSvcName=webhooks-00038 old-service=nil service="frontends:[172.23.27.80]/ports=[http]/selector=map[]" subsys=k8s-watcher
level=debug msg="Upserting service" backends="[{http 0 borin6 {10.2.4.20 {TCP 8012} 0} 0 false false}]" l7LBFrontendPorts="[]" l7LBProxyPort=0 loadBalancerSourceRanges="[]" serviceIP="{172.23.27.80 {TCP 80} 0}" serviceName=webhooks-00038 serviceNamespace=sa-common sessionAffinity=false sessionAffinityTimeout=0 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=ClusterIP
level=debug msg="Acquired service ID" backends="[{http 0 borin6 {10.2.4.20 {TCP 8012} 0} 0 false false}]" l7LBFrontendPorts="[]" l7LBProxyPort=0 loadBalancerSourceRanges="[]" serviceID=506 serviceIP="{172.23.27.80 {TCP 80} 0}" serviceName=webhooks-00038 serviceNamespace=sa-common sessionAffinity=false sessionAffinityTimeout=0 subsys=service svcHealthCheckNodePort=0 svcTrafficPolicy=Cluster svcType=ClusterIP
level=debug msg="Deleting backends from session affinity match" backends="[]" serviceID=506 subsys=service
level=debug msg="Upserted service entry" backendSlot=1 subsys=map-lb svcKey="172.23.27.80:20480" svcVal="1 0 (64001) [0x0 0x0]"
level=debug msg="Upserted service entry" backendSlot=0 subsys=map-lb svcKey="172.23.27.80:20480" svcVal="0 1 (64001) [0x0 0x0]"
level=debug msg="Adding backends to affinity match map" backends="[]" serviceID=506 subsys=service
level=debug msg="Kubernetes service definition changed" action=service-updated endpoints="10.2.2.148:10901/TCP,10.2.2.148:10902/TCP" k8sNamespace=sa k8sSvcName=thanos-store-0 old-service=nil service="frontends:[]/ports=[grpc http]/selector=map[app.kubernetes.io/component:object-store-gateway app.kubernetes.io/instance:thanos-store-0 app.kubernetes.io/name:thanos-store store.thanos.io/shard:shard-0]" subsys=k8s-watcher
level=debug msg="Starting new controller" name="propagating local node change to kv-store" subsys=controller uuid=f7c03876-2a4e-48d2-9e04-1fe9b4e07006
level=debug msg="Controller func execution time: 1.329597ms" name="propagating local node change to kv-store" subsys=controller uuid=f7c03876-2a4e-48d2-9e04-1fe9b4e07006
level=debug msg="Controller run succeeded; waiting for next controller update or stop" name="propagating local node change to kv-store" subsys=controller uuid=f7c03876-2a4e-48d2-9e04-1fe9b4e07006
level=debug msg="Received event from etcd: {Header:{ClusterId:8482860168806118984 MemberId:8744597047382772316 Revision:1007713559 RaftTerm:46 XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0} Events:[0xc000faa900] CompactRevision:0 Canceled:false Created:false closeErr:<nil> cancelReason:}" config=/var/lib/etcd-config/etcd.config endpoints="[https://etcd3-sa-k8s-cvm.svc.douban:10079 https://etcd1-sa-k8s-cvm.svc.douban:10079 https://etcd2-sa-k8s-cvm.svc.douban:10079]" prefix=cilium/state/nodes/v1 revision=1007713559 subsys=kvstore watcher=store-cilium/state/nodes/v1-watcher
level=debug msg="Emitting modify event for cilium/state/nodes/v1/default/bullseye1=[123 34 78 97 109 101 34 58 34 98 117 108 108 115 101 121 101 49 34 44 34 67 108 117 115 116 101 114 34 58 34 100 101 102 97 117 108 116 34 44 34 73 80 65 100 100 114 101 115 115 101 115 34 58 91 123 34 84 121 112 101 34 58 34 73 110 116 101 114 110 97 108 73 80 34 44 34 73 80 34 58 34 49 48 46 49 46 51 50 46 57 55 34 125 44 123 34 84 121 112 101 34 58 34 67 105 108 105 117 109 73 110 116 101 114 110 97 108 73 80 34 44 34 73 80 34 58 34 49 48 46 50 46 53 46 54 51 34 125 93 44 34 73 80 118 52 65 108 108 111 99 67 73 68 82 34 58 123 34 73 80 34 58 34 49 48 46 50 46 53 46 48 34 44 34 77 97 115 107 34 58 34 47 47 47 47 65 65 61 61 34 125 44 34 73 80 118 52 83 101 99 111 110 100 97 114 121 65 108 108 111 99 67 73 68 82 115 34 58 110 117 108 108 44 34 73 80 118 54 65 108 108 111 99 67 73 68 82 34 58 110 117 108 108 44 34 73 80 118 54 83 101 99 111 110 100 97 114 121 65 108 108 111 99 67 73 68 82 115 34 58 110 117 108 108 44 34 73 80 118 52 72 101 97 108 116 104 73 80 34 58 34 49 48 46 50 46 53 46 49 55 56 34 44 34 73 80 118 54 72 101 97 108 116 104 73 80 34 58 34 34 44 34 73 80 118 52 73 110 103 114 101 115 115 73 80 34 58 34 34 44 34 73 80 118 54 73 110 103 114 101 115 115 73 80 34 58 34 34 44 34 67 108 117 115 116 101 114 73 68 34 58 48 44 34 83 111 117 114 99 101 34 58 34 108 111 99 97 108 34 44 34 69 110 99 114 121 112 116 105 111 110 75 101 121 34 58 48 44 34 76 97 98 101 108 115 34 58 123 34 98 101 116 97 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 97 114 99 104 34 58 34 97 109 100 54 52 34 44 34 98 101 116 97 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 111 115 34 58 34 108 105 110 117 120 34 44 34 107 117 98 101 114 110 101 116 101 115 46 105 111 47 97 114 99 104 34 58 34 97 109 100 54 52 34 44 34 107 117 98 101 114 110 101 116 101 115 46 105 111 47 104 111 115 116 110 97 109 101 34 58 34 98 117 108 108 115 101 121 101 49 34 44 34 107 117 98 101 114 110 101 116 101 115 46 105 111 47 111 115 34 58 34 108 105 110 117 120 34 44 34 116 111 112 111 108 111 103 121 46 99 111 109 46 116 101 110 99 101 110 116 46 99 108 111 117 100 46 99 115 105 46 99 98 115 47 122 111 110 101 34 58 34 97 112 45 98 101 105 106 105 110 103 45 53 34 125 44 34 78 111 100 101 73 100 101 110 116 105 116 121 34 58 49 44 34 87 105 114 101 103 117 97 114 100 80 117 98 75 101 121 34 58 34 34 125]" config=/var/lib/etcd-config/etcd.config endpoints="[https://etcd3-sa-k8s-cvm.svc.douban:10079 https://etcd1-sa-k8s-cvm.svc.douban:10079 https://etcd2-sa-k8s-cvm.svc.douban:10079]" prefix=cilium/state/nodes/v1 revision=1007713559 subsys=kvstore watcher=store-cilium/state/nodes/v1-watcher
level=debug msg="Received key update via kvstore [value {\"Name\":\"bullseye1\",\"Cluster\":\"default\",\"IPAddresses\":[{\"Type\":\"InternalIP\",\"IP\":\"10.1.32.97\"},{\"Type\":\"CiliumInternalIP\",\"IP\":\"10.2.5.63\"}],\"IPv4AllocCIDR\":{\"IP\":\"10.2.5.0\",\"Mask\":\"////AA==\"},\"IPv4SecondaryAllocCIDRs\":null,\"IPv6AllocCIDR\":null,\"IPv6SecondaryAllocCIDRs\":null,\"IPv4HealthIP\":\"10.2.5.178\",\"IPv6HealthIP\":\"\",\"IPv4IngressIP\":\"\",\"IPv6IngressIP\":\"\",\"ClusterID\":0,\"Source\":\"local\",\"EncryptionKey\":0,\"Labels\":{\"beta.kubernetes.io/arch\":\"amd64\",\"beta.kubernetes.io/os\":\"linux\",\"kubernetes.io/arch\":\"amd64\",\"kubernetes.io/hostname\":\"bullseye1\",\"kubernetes.io/os\":\"linux\",\"topology.com.tencent.cloud.csi.cbs/zone\":\"ap-beijing-5\"},\"NodeIdentity\":1,\"WireguardPubKey\":\"\"}]" eventType=modify key=cilium/state/nodes/v1/default/bullseye1 storeName=store-cilium/state/nodes/v1 subsys=shared-store
level=debug msg="Received node update event from kvstore: types.Node{Name:\"bullseye1\", Cluster:\"default\", IPAddresses:[]types.Address{types.Address{Type:\"InternalIP\", IP:net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xa, 0x1, 0x20, 0x61}}, types.Address{Type:\"CiliumInternalIP\", IP:net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xa, 0x2, 0x5, 0x3f}}}, IPv4AllocCIDR:(*cidr.CIDR)(0xc001dfa260), IPv4SecondaryAllocCIDRs:[]*cidr.CIDR(nil), IPv6AllocCIDR:(*cidr.CIDR)(nil), IPv6SecondaryAllocCIDRs:[]*cidr.CIDR(nil), IPv4HealthIP:net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xa, 0x2, 0x5, 0xb2}, IPv6HealthIP:net.IP(nil), IPv4IngressIP:net.IP(nil), IPv6IngressIP:net.IP(nil), ClusterID:0, Source:\"kvstore\", EncryptionKey:0x0, Labels:map[string]string{\"beta.kubernetes.io/arch\":\"amd64\", \"beta.kubernetes.io/os\":\"linux\", \"kubernetes.io/arch\":\"amd64\", \"kubernetes.io/hostname\":\"bullseye1\", \"kubernetes.io/os\":\"linux\", \"topology.com.tencent.cloud.csi.cbs/zone\":\"ap-beijing-5\"}, NodeIdentity:0x1, WireguardPubKey:\"\"}" subsys=nodemanager
level=debug msg="Controller func execution time: 113.997µs" name=metricsmap-bpf-prom-sync subsys=controller uuid=2f7bce2a-72f3-47f1-a884-6bc69e7d6e57
level=info msg="Exiting due to signal" signal=terminated subsys=daemon
level=error msg="Failed to start the udp DNS proxy on [::]:40819" error="read udp [::]:40819: use of closed network connection" subsys=fqdn/dnsproxy
level=error msg="Failed to start the tcp DNS proxy on [::]:40819" error="accept tcp [::]:40819: use of closed network connection" subsys=fqdn/dnsproxy
level=debug msg="canceling context in signal handler" subsys=daemon
level=info msg="Waiting for all endpoints' go routines to be stopped." subsys=daemon
level=info msg="All endpoints' goroutines stopped." subsys=daemon
level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state /host/proc/sys/net /sys/class/net 10.2.5.63 <nil> tunnel vxlan 8472 eth0 cilium_host cilium_net 1500 true true true /run/cilium/cgroupv2 /sys/fs/bpf true true v3 2 false true 1]" error="signal: killed" subsys=datapath-loader
level=fatal msg="Error while creating daemon" error="error while initializing daemon: failed while reinitializing datapath: Command execution failed for [/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state /host/proc/sys/net /sys/class/net 10.2.5.63 <nil> tunnel vxlan 8472 eth0 cilium_host cilium_net 1500 true true true /run/cilium/cgroupv2 /sys/fs/bpf true true v3 2 false true 1]: context canceled" subsys=daemon
```

### Anything else?

- It  works very well on buster node with exactly same config and kernel version
- I have set all the `rp_filter` sysctl to 0
- containerd/crun/cni-plugin have upgrade to the newest version
- Node doesn't have disk/memory pressure or any OOM events
- descibe pod
```
> kubectl describe pod -n kube-system cilium-8dbzk
Name:                 cilium-8dbzk
Namespace:            kube-system
Priority:             2000001000
Priority Class Name:  system-node-critical
Node:                 bullseye1/10.1.32.97
Start Time:           Wed, 28 Sep 2022 17:33:25 +0800
Labels:               controller-revision-hash=d8c849567
                      k8s-app=cilium
                      pod-template-generation=16
Annotations:          container.apparmor.security.beta.kubernetes.io/apply-sysctl-overwrites: unconfined
                      container.apparmor.security.beta.kubernetes.io/cilium-agent: unconfined
                      container.apparmor.security.beta.kubernetes.io/clean-cilium-state: unconfined
                      container.apparmor.security.beta.kubernetes.io/mount-cgroup: unconfined
                      kubectl.kubernetes.io/restartedAt: 2021-02-22T12:02:15+08:00
Status:               Running
IP:                   10.1.32.97
IPs:
  IP:           10.1.32.97
Controlled By:  DaemonSet/cilium
Init Containers:
  mount-cgroup:
    Container ID:  containerd://c28f8147d62668a3870b8f2f8b6972328c257d7bb0ebc90b2889eee26cf7a2b7
    Image:         quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Image ID:      quay.io/cilium/cilium@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -ec
      cp /usr/bin/cilium-mount /hostbin/cilium-mount;
      nsenter --cgroup=/hostproc/1/ns/cgroup --mount=/hostproc/1/ns/mnt "${BIN_PATH}/cilium-mount" $CGROUP_ROOT;
      rm /hostbin/cilium-mount
      
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 28 Sep 2022 17:38:00 +0800
      Finished:     Wed, 28 Sep 2022 17:38:00 +0800
    Ready:          True
    Restart Count:  2
    Environment:
      CGROUP_ROOT:  /run/cilium/cgroupv2
      BIN_PATH:     /opt/cni/bin
    Mounts:
      /hostbin from cni-path (rw)
      /hostproc from hostproc (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-798ff (ro)
  apply-sysctl-overwrites:
    Container ID:  containerd://54415ae76286b0daa181664f5baa8aae0148158dbb7a2768b40f8226c74053c4
    Image:         quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Image ID:      quay.io/cilium/cilium@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -ec
      cp /usr/bin/cilium-sysctlfix /hostbin/cilium-sysctlfix;
      nsenter --mount=/hostproc/1/ns/mnt "${BIN_PATH}/cilium-sysctlfix";
      rm /hostbin/cilium-sysctlfix
      
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 28 Sep 2022 17:38:00 +0800
      Finished:     Wed, 28 Sep 2022 17:38:00 +0800
    Ready:          True
    Restart Count:  0
    Environment:
      BIN_PATH:  /opt/cni/bin
    Mounts:
      /hostbin from cni-path (rw)
      /hostproc from hostproc (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-798ff (ro)
  mount-bpf-fs:
    Container ID:  containerd://e7ec493720071a985d195e4343dab76718efd4928b0fbf63945e025f2e694fbd
    Image:         quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Image ID:      quay.io/cilium/cilium@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/bash
      -c
      --
    Args:
      mount | grep "/sys/fs/bpf type bpf" || mount -t bpf bpf /sys/fs/bpf
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 28 Sep 2022 17:38:01 +0800
      Finished:     Wed, 28 Sep 2022 17:38:02 +0800
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /sys/fs/bpf from bpf-maps (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-798ff (ro)
  clean-cilium-state:
    Container ID:  containerd://38f8cb96b847f697a376b9c5f44953425e91dd4a74e6dea01028e5b6ef75465c
    Image:         quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Image ID:      quay.io/cilium/cilium@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Port:          <none>
    Host Port:     <none>
    Command:
      /init-container.sh
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Wed, 28 Sep 2022 17:38:02 +0800
      Finished:     Wed, 28 Sep 2022 17:38:02 +0800
    Ready:          True
    Restart Count:  0
    Requests:
      cpu:     100m
      memory:  100Mi
    Environment:
      CILIUM_ALL_STATE:         <set to the key 'clean-cilium-state' of config map 'cilium-config'>      Optional: true
      CILIUM_BPF_STATE:         <set to the key 'clean-cilium-bpf-state' of config map 'cilium-config'>  Optional: true
      KUBERNETES_SERVICE_HOST:  sa-borin.k8s.douban
      KUBERNETES_SERVICE_PORT:  443
    Mounts:
      /run/cilium/cgroupv2 from cilium-cgroup (rw)
      /sys/fs/bpf from bpf-maps (rw)
      /var/run/cilium from cilium-run (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-798ff (ro)
Containers:
  cilium-agent:
    Container ID:  containerd://8c9c236ca68e8a4d8af8a04f6ca2ae3c37d4ec80f0f056330c291d628b8a8be8
    Image:         quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Image ID:      quay.io/cilium/cilium@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36
    Port:          <none>
    Host Port:     <none>
    Command:
      cilium-agent
    Args:
      --debug
      --config-dir=/tmp/cilium/config-map
    State:       Waiting
      Reason:    CrashLoopBackOff
    Last State:  Terminated
      Reason:    Error
      Message:   28673) [0x0 0x0]"
level=debug msg="Upserted service entry" backendSlot=0 subsys=map-lb svcKey="0.0.0.0:13435" svcVal="0 3 (28673) [0x2 0x0]"
level=debug msg="Adding backends to affinity match map" backends="[]" serviceID=368 subsys=service
level=debug msg="cache synced" kubernetesResource="core/v1::Service" subsys=k8s
level=debug msg="stopped waiting for caches to be synced" kubernetesResource="core/v1::Service" subsys=k8s
level=debug msg="resource \"core/v1::Service\" cache has synced, stopping timeout watcher" subsys=k8s
level=debug msg="Controller func execution time: 91.814µs" name=metricsmap-bpf-prom-sync subsys=controller uuid=9f671838-7029-4a74-bc13-38d21ed290a5
level=info msg="Exiting due to signal" signal=terminated subsys=daemon
level=error msg="Failed to start the udp DNS proxy on [::]:40819" error="read udp [::]:40819: use of closed network connection" subsys=fqdn/dnsproxy
level=error msg="Failed to start the tcp DNS proxy on [::]:40819" error="accept tcp [::]:40819: use of closed network connection" subsys=fqdn/dnsproxy
level=debug msg="canceling context in signal handler" subsys=daemon
level=info msg="Waiting for all endpoints' go routines to be stopped." subsys=daemon
level=info msg="All endpoints' goroutines stopped." subsys=daemon
level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state /host/proc/sys/net /sys/class/net 10.2.5.63 <nil> tunnel vxlan 8472 eth0 cilium_host cilium_net 1500 true true true /run/cilium/cgroupv2 /sys/fs/bpf true true v3 2 false true 1]" error="signal: killed" subsys=datapath-loader
level=fatal msg="Error while creating daemon" error="error while initializing daemon: failed while reinitializing datapath: Command execution failed for [/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state /host/proc/sys/net /sys/class/net 10.2.5.63 <nil> tunnel vxlan 8472 eth0 cilium_host cilium_net 1500 true true true /run/cilium/cgroupv2 /sys/fs/bpf true true v3 2 false true 1]: context canceled" subsys=daemon

      Exit Code:    137
      Started:      Wed, 28 Sep 2022 17:37:48 +0800
      Finished:     Wed, 28 Sep 2022 17:37:57 +0800
    Ready:          False
    Restart Count:  5
    Liveness:       http-get http://127.0.0.1:9879/healthz delay=0s timeout=5s period=30s #success=1 #failure=10
    Readiness:      http-get http://127.0.0.1:9879/healthz delay=0s timeout=5s period=30s #success=1 #failure=3
    Startup:        http-get http://127.0.0.1:9879/healthz delay=0s timeout=1s period=30s #success=1 #failure=105
    Environment:
      K8S_NODE_NAME:               (v1:spec.nodeName)
      CILIUM_K8S_NAMESPACE:       kube-system (v1:metadata.namespace)
      CILIUM_CLUSTERMESH_CONFIG:  /var/lib/cilium/clustermesh/
      CILIUM_CNI_CHAINING_MODE:   <set to the key 'cni-chaining-mode' of config map 'cilium-config'>  Optional: true
      CILIUM_CUSTOM_CNI_CONF:     <set to the key 'custom-cni-conf' of config map 'cilium-config'>    Optional: true
      KUBERNETES_SERVICE_HOST:    sa-borin.k8s.douban
      KUBERNETES_SERVICE_PORT:    443
    Mounts:
      /host/etc/cni/net.d from etc-cni-netd (rw)
      /host/opt/cni/bin from cni-path (rw)
      /host/proc/sys/kernel from host-proc-sys-kernel (rw)
      /host/proc/sys/net from host-proc-sys-net (rw)
      /lib/modules from lib-modules (ro)
      /run/xtables.lock from xtables-lock (rw)
      /sys/fs/bpf from bpf-maps (rw)
      /tmp/cilium/config-map from cilium-config-path (ro)
      /var/lib/cilium/clustermesh from clustermesh-secrets (ro)
      /var/lib/cilium/tls/hubble from hubble-tls (ro)
      /var/lib/etcd-config from etcd-config-path (ro)
      /var/lib/etcd-secrets from etcd-secrets (ro)
      /var/run/cilium from cilium-run (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-798ff (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  cilium-run:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/cilium
    HostPathType:  DirectoryOrCreate
  bpf-maps:
    Type:          HostPath (bare host directory volume)
    Path:          /sys/fs/bpf
    HostPathType:  DirectoryOrCreate
  hostproc:
    Type:          HostPath (bare host directory volume)
    Path:          /proc
    HostPathType:  Directory
  cilium-cgroup:
    Type:          HostPath (bare host directory volume)
    Path:          /run/cilium/cgroupv2
    HostPathType:  DirectoryOrCreate
  cni-path:
    Type:          HostPath (bare host directory volume)
    Path:          /opt/cni/bin
    HostPathType:  DirectoryOrCreate
  etc-cni-netd:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/cni/net.d
    HostPathType:  DirectoryOrCreate
  lib-modules:
    Type:          HostPath (bare host directory volume)
    Path:          /lib/modules
    HostPathType:  
  xtables-lock:
    Type:          HostPath (bare host directory volume)
    Path:          /run/xtables.lock
    HostPathType:  FileOrCreate
  etcd-config-path:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cilium-config
    Optional:  false
  etcd-secrets:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  cilium-etcd-secrets
    Optional:    true
  clustermesh-secrets:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  cilium-clustermesh
    Optional:    true
  cilium-config-path:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cilium-config
    Optional:  false
  host-proc-sys-net:
    Type:          HostPath (bare host directory volume)
    Path:          /proc/sys/net
    HostPathType:  Directory
  host-proc-sys-kernel:
    Type:          HostPath (bare host directory volume)
    Path:          /proc/sys/kernel
    HostPathType:  Directory
  hubble-tls:
    Type:                Projected (a volume that contains injected data from multiple sources)
    SecretName:          hubble-server-certs
    SecretOptionalName:  0xc00073584a
  kube-api-access-798ff:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              kubernetes.io/os=linux
Tolerations:                 op=Exists
                             node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason          Age                    From               Message
  ----     ------          ----                   ----               -------
  Normal   Scheduled       6m54s                  default-scheduler  Successfully assigned kube-system/cilium-8dbzk to bullseye1
  Normal   Started         6m52s                  kubelet            Started container mount-bpf-fs
  Normal   Started         6m51s                  kubelet            Started container clean-cilium-state
  Normal   Pulled          6m51s                  kubelet            Container image "quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36" already present on machine
  Normal   Created         6m51s                  kubelet            Created container clean-cilium-state
  Normal   Started         6m50s                  kubelet            Started container cilium-agent
  Normal   Created         6m50s                  kubelet            Created container cilium-agent
  Normal   Pulled          6m50s                  kubelet            Container image "quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36" already present on machine
  Normal   Killing         6m24s                  kubelet            Stopping container cilium-agent
  Normal   Started         6m22s (x2 over 6m54s)  kubelet            Started container mount-cgroup
  Normal   Pulled          6m22s (x2 over 6m54s)  kubelet            Container image "quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36" already present on machine
  Normal   SandboxChanged  6m22s                  kubelet            Pod sandbox changed, it will be killed and re-created.
  Normal   Created         6m22s (x2 over 6m54s)  kubelet            Created container mount-cgroup
  Normal   Started         6m21s (x2 over 6m53s)  kubelet            Started container apply-sysctl-overwrites
  Normal   Created         6m21s (x2 over 6m53s)  kubelet            Created container apply-sysctl-overwrites
  Normal   Pulled          6m21s (x2 over 6m53s)  kubelet            Container image "quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36" already present on machine
  Normal   Created         6m20s (x2 over 6m52s)  kubelet            Created container mount-bpf-fs
  Normal   Pulled          6m20s (x2 over 6m52s)  kubelet            Container image "quay.io/cilium/cilium:v1.12.2@sha256:986f8b04cfdb35cf714701e58e35da0ee63da2b8a048ab596ccb49de58d5ba36" already present on machine
  Warning  BackOff         114s (x22 over 5m49s)  kubelet            Back-off restarting failed container
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dispensable** [comment]
Cilium host routing is works. I can ping 10.2.5.63(bullseye cilium_host) to other buster node and visa versa. The /healthz probe can be curl from bullseye host with curl $hostname:port/healthz with 200 resp for a short time. 
```
> ip route 
default via 10.1.32.1 dev eth0 
10.0.0.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.0.3.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.1.32.0/24 dev eth0 proto kernel scope link src 10.1.32.97 
10.2.0.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.1.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.2.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.3.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.4.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.5.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 
10.2.5.63 dev cilium_host scope link 
10.2.6.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.7.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.8.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450 
10.2.10.0/24 via 10.2.5.63 dev cilium_host src 10.2.5.63 mtu 1450
```

And `rp_filter` settings:

```
> sudo /sbin/sysctl -a | grep rp_filter
net.ipv4.conf.all.arp_filter = 0
net.ipv4.conf.all.rp_filter = 0
net.ipv4.conf.cilium_host.arp_filter = 0
net.ipv4.conf.cilium_host.rp_filter = 0
net.ipv4.conf.cilium_net.arp_filter = 0
net.ipv4.conf.cilium_net.rp_filter = 0
net.ipv4.conf.cilium_vxlan.arp_filter = 0
net.ipv4.conf.cilium_vxlan.rp_filter = 0
net.ipv4.conf.default.arp_filter = 0
net.ipv4.conf.default.rp_filter = 0
net.ipv4.conf.eth0.arp_filter = 0
net.ipv4.conf.eth0.rp_filter = 0
net.ipv4.conf.lo.arp_filter = 0
net.ipv4.conf.lo.rp_filter = 0
```

**Comment by dispensable** [comment]
Update:

We reconfigured containerd's config to ensure `crun` enable `cgroupv2` support. And it works very well now. 

Thank you anyway
